File: 2019/day-06/part2.rb
def trace(orbital_map, k)
  return [] unless orbital_map.key?(k)

  [orbital_map[k]] + trace(orbital_map, orbital_map[k])
end

def process(data)
  source_trace = trace(data, 'YOU')
  destination_trace = trace(data, 'SAN')

  crosspoint = (source_trace & destination_trace).first
  source_trace.index(crosspoint) + destination_trace.index(crosspoint)
end

def input
  File.readlines('./input.txt').map do |line|
    line.strip.split(')').reverse
  end.to_h
end

puts process(input)

File: 2015/day-02/part2.py
import math

def input():
  return list(
    map(
      lambda line: [int(d) for d in line.strip().split('x')],
      open('./input.txt', 'r').readlines()
    )
  )

def required_ribbon(dimensions):
  return 2 * sum(sorted(dimensions)[:2]) + math.prod(dimensions)

def process(data):
  return sum(map(required_ribbon, data))

print(process(input()))
